numpy/core/src/_simd/_simd_inc_vector.h

```diff
@@ -0,0 +1,206 @@
+#ifndef NPY_SIMD
+    #error "Not a standalone header"
+#endif
+
+/************************************
+ ** Private Definitions
+ ************************************/
+// PySequenceMethods
+static Py_ssize_t
+simd__vector_length(simd_vector *self)
+{
+    return simd_data_getinfo(self->type)->nlanes;
+}
+static PyObject *
+simd__vector_item(simd_vector *self, Py_ssize_t i)
+{
+    const simd_data_info *info = simd_data_getinfo(self->type);
+    int nlanes = info->nlanes;
+    if (i >= nlanes) {
+        PyErr_SetString(PyExc_IndexError, "list index out of range");
+        return NULL;
+    }
+    simd_data d;
+    d.qu8 = self->data + i * info->lane_size;
+    d.u64 = *d.qu64;
```

Comment:
This is super scary to me. You have a union containing both a `T` and a `T*`, and then you swap between them.

What would change if you split this union into `simd_data` and `simd_dataq`? Are there any bits of this file that actually need the union to mix pointers and non-pointers?

Reply:
> This is super scary to me. You have a union containing both a T and a T*, and then you swap between them.

LGTM, I swap between two pointers(reinterpret cast) then deference it to another item and the origin pointer is aligned to over 64bit.

> What would change if you split this union into simd_data and simd_dataq?

`q*` types represent sequences types, and pointers should be gathered with the other types in one union
to simplify the implementations of python methods that carry the NPYV intrinsics, check macros in `_simd_inc_easyintrin.h` and how we use them in `_simd.dispatch.c.src`.

Reply:
> I swap between two pointers(reinterpret cast)
There's no need to reinterpret-cast pointers like this, a static-cast would do just fine. Doing all this work in the same local variable is what makes things needlessly confusing. Why not:
```suggestion
    simd_data d = {.u64 = *(npyv_lanetype_u64 *)(self->data + i * info->lane_size)};
```

Reply:
static-cast may raise, strict-aliasing error by the compiler, I will give it a try.